From d4283e9db58ed7b6fe6265d410bbbd2a818f85d4 Mon Sep 17 00:00:00 2001
Message-Id: <d4283e9db58ed7b6fe6265d410bbbd2a818f85d4.1297957056.git.jdenemar@redhat.com>
From: Daniel P. Berrange <berrange@redhat.com>
Date: Tue, 15 Feb 2011 17:27:31 +0000
Subject: [PATCH] Fix typo in setting up SPICE passwords

For

https://bugzilla.redhat.com/show_bug.cgi?id=677709

A copy & paste typo caused SPICE password setup to reference
the wrong part of a union. Previously this was harmless because
both spice & vnc had their 'auth' fields at the same offset.
The addition of UNIX socket support to VNC moved the auth
field exposing the bug

* src/qemu/qemu_driver.c: s/vnc/spice/ in password setup

This fix is already included in upstream commit
9d73efdbe3ea61a13a11fdc24a2cb530eaa0b66f.
---
 src/qemu/qemu_driver.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qemu/qemu_driver.c b/src/qemu/qemu_driver.c
index 3cdb462..e2a301a 100644
--- a/src/qemu/qemu_driver.c
+++ b/src/qemu/qemu_driver.c
@@ -2405,8 +2405,8 @@ qemuInitPasswords(virConnectPtr conn,
         } else if (vm->def->graphics[0]->type == VIR_DOMAIN_GRAPHICS_TYPE_SPICE) {
             ret = qemuInitGraphicsPasswords(driver, vm,
                                             VIR_DOMAIN_GRAPHICS_TYPE_SPICE,
-                                            &vm->def->graphics[0]->data.vnc.auth,
-                                            driver->vncPassword);
+                                            &vm->def->graphics[0]->data.spice.auth,
+                                            driver->spicePassword);
         }
     }
 
-- 
1.7.4.1

